pkg/asset/agent/mirror/registriesconf.go

```diff
@@ -208,6 +214,20 @@ func (i *RegistriesConf) finish() error {
 	return nil
 }
 
+func (i *RegistriesConf) validateReleaseImageIsSameInRegistriesConf() error {
+
+	releaseImage := &releaseimage.Image{}
+	releaseImage.Generate(asset.Parents{})
+
+	releaseImagePath := strings.Split(releaseImage.PullSpec, ":")[0]
+
+	if !strings.Contains(string(i.File.Data), releaseImagePath) {
+		return errors.New(fmt.Sprintf("%s should have an entry matching the releaseImage %s", RegistriesConfFilename, releaseImagePath))
```

Comment:
It's tricky to know what to put as the source here, since the user may have started with either an ICSP in the InstallConfig (when called from `Generate()`) or from a mirror/registries.conf file (when called from `Load()`). Ideally we could distinguish between the two cases.